### PR TITLE
Ignore hidden projects

### PR DESCRIPTION
**Pull request by Rexios80**
Fixes https://github.com/invertase/dart_custom_lint/issues/271

### CODE REVIEW

File: packages/custom_lint/lib/src/workspace.dart [hunk 1]
```diff
@@ -424,6 +424,11 @@ Iterable<String> _findRoots(String path) sync* {
   final directory = Directory(path);
 
   yield* directory.listSync(recursive: true).whereType<File>().where((file) {
+    final relativePathSegments = split(relative(file.path));
```
Comment (rrousselGit) [inline packages/custom_lint/lib/src/workspace.dart:427]: I think the correct fix is to remove the `isChildFromCache` logic and always return `packageConfig.existsSync`

**Review by rrousselGit — CHANGES_REQUESTED** [review]
Thanks! We'd need a test though :)

File: packages/custom_lint/lib/src/workspace.dart (hunk 1 above)
Comment (Rexios80) [reply packages/custom_lint/lib/src/workspace.dart:427]: This seems to work. Thanks!

**Comment by Rexios80** [comment]
Do we still need a test for this? Is there an existing test I can modify?

**Comment by rrousselGit** [comment]
We always need tests :)

You can modify the one that was added in the previous attempt at fixing this issue: https://github.com/invertase/dart_custom_lint/commit/773b7ef69b0ddae77cbfc803364fdbe8bf4c6846

**Comment by Rexios80** [comment]
@rrousselGit I updated the test and double checked that it fails without the fix

**Comment by rrousselGit** [comment]
I'll fix the CI tomorrow. We'll then see if this PR doesn't cause any regression

**Comment by CPRmichel** [comment]
any updates on this matter?

**Comment by rrousselGit** [comment]
Dang it, forgot.
master's CI is fixed. But it appears that this PR causes some tests to fail

**Comment by Rexios80** [comment]
@rrousselGit The test that's failing is checking if an exception is thrown when `package_config.json` is missing. The purpose of this PR is to _not_ crash when that is the case. Should we remove that test?

**Comment by rrousselGit** [comment]
Hum, I think we can remove it yes. It should be tested separately by your test update.

I wonder what that `workspace_test.dart` failure is about though.

**Comment by Rexios80** [comment]
Maybe it used to expect a project to be found that is missing the package file?

**Comment by rrousselGit** [comment]
It looks like `analysis_options.yaml` on their own are no-longer detected.

The test name is a bit misleading as it has nothing to do with pubspecs afaik.

**Comment by Rexios80** [comment]
@rrousselGit The tests should pass now

**Comment by rrousselGit** [comment]
Thanks!

**Pull request merged**